### PR TITLE
Nested Interrupts: port to new architecture

### PR DESCRIPTION
**Pull request by nfcampos**

### CODE REVIEW

File: libs/langgraph/langgraph/pregel/loop.py [hunk 1]
```diff
@@ -348,6 +370,20 @@ def __exit__(
...
+            if exc_value.args[0] is self:
+                # interrupt raised by this loop
+                exc_value.args = (object(),)
+            else:
+                # interrupt raised by a nested loop, save interrupt checkpoint
+                self._put_checkpoint({"source": "interrupt"})
+            if not self.is_nested:
+                # in outer graph, catch interrupt
+                del self.graph
+                return True and self.stack.__exit__(None, None, None)
```
Comment (nfcampos) [inline libs/langgraph/langgraph/pregel/loop.py:384]:
```suggestion
                return True or self.stack.__exit__(None, None, None)
```

File: libs/langgraph/langgraph/pregel/loop.py [hunk 2]
```diff
@@ -417,6 +454,22 @@ async def __aexit__(
...
+            if exc_value.args[0] is self:
+                # interrupt raised by this loop
+                exc_value.args = (object(),)
+            else:
+                # interrupt raised by a nested loop, save interrupt checkpoint
+                self._put_checkpoint({"source": "interrupt"})
+            if not self.is_nested:
+                # in outer graph, catch interrupt
+                del self.graph
+                return True and await asyncio.shield(
```
Comment (nfcampos) [inline libs/langgraph/langgraph/pregel/loop.py:468]:
```suggestion
                return True or await asyncio.shield(
```

File: libs/langgraph/langgraph/pregel/__init__.py [hunk 3]
```diff
@@ -1201,6 +1221,7 @@ async def astream(
                     read_channels(loop.channels, output_keys)
                 )
         except BaseException as e:
+            # TODO use on_chain_end if exc is GraphInterrupt
```
Comment (nfcampos) [inline libs/langgraph/langgraph/pregel/__init__.py:1224]: this still needs to be done

File: libs/langgraph/langgraph/pregel/executor.py [hunk 4]
```diff
@@ -79,7 +102,7 @@ def __init__(self) -> None:
 
     def submit(
         self,
-        fn: Callable[P, Awaitable[T]],
```
Comment (nfcampos) [inline libs/langgraph/langgraph/pregel/executor.py:82]: why is this changing @vbarda ?

File: libs/langgraph/langgraph/pregel/executor.py [hunk 5]
```diff
@@ -34,41 +35,63 @@ def __call__(
         ...
 
 
-@contextmanager
-def BackgroundExecutor(config: RunnableConfig) -> Iterator[Submit]:
-    tasks: dict[concurrent.futures.Future, bool] = {}
-    with get_executor_for_config(config) as executor:
+class BackgroundExecutor(ContextManager):
```
Comment (nfcampos) [inline libs/langgraph/langgraph/pregel/executor.py:38]: this rewrites BackgroundExecutor to wait all tasks before exit, like the async one

File: libs/langgraph/langgraph/pregel/executor.py (hunk 4 above)
Comment (vbarda) [reply libs/langgraph/langgraph/pregel/executor.py:82]: don't think this is from my change lol

**Pull request merged**